src/writer/async_tokio.rs

```diff
@@ -339,4 +385,46 @@ mod indentation_async {
 </paired>"#
         );
     }
+
+    #[tokio::test]
+    async fn element_writer_nested() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
```

Comment:
I'm fairly certain that the only reason this works is that you're using a buffer type which also implements `Write` rather than an exclusively-async type. 

`write_text_content` and other methods aren't implemented for `ElementWriter<'a, W: AsyncWrite + Unpin>`

As a sidenote if you can avoid duplicating the definition of `ElementWriter`, please do.

Reply:
> I'm fairly certain that the only reason this works is that you're using a buffer type which also implements Write rather than an exclusively-async type.

Good catch! I'll update that

> As a sidenote if you can avoid duplicating the definition of ElementWriter, please do.

AFAIK, the only way to do that would be to have an async-only implementation, then have the sync implementation be a wrapper around it. But since async is an optional feature in this crate (and perf is important), I don't think that's the best option. 

If there's a better way, I'd love to know! :)

Reply:
You could remove `Write` requirement from `ElementWriter` struct and from impl block for attributes and put `Write` or `AsyncWrite + Unpin` to individual impl blocks. Then new functions should be named with `_async` suffix and no need separate method to create async element writer.

Reply:
That's a good idea, I like that structure a lot more!